src/de.rs

```diff
@@ -229,8 +236,11 @@ impl<'de, 'a, R: Read> de::Deserializer<'de> for &'a mut Deserializer<R> {
     where
         V: Visitor<'de>,
     {
-        let len = self.rdr.bin_read_nat0()?;
-        visitor.visit_map(SeqAccess::new(self, len))
+        // we can't know the field names (and don't need to) if we are deserializing in
+        // stronly typed mode. To make everything work just add some dummy field names
+        let len: usize = self.rdr.bin_read_nat0()?;
+        let dummy_fields = std::iter::repeat("".to_string()).take(len).collect();
```

Comment:
is this parsing: 
```
"ref_rule": [ "Record", ...
``` 
section? If yes, we do have record field names there

Reply:
oh wait, this is reading the binary block; right?

Reply:
I modified the `MapAccess` visitor to accept a list of field names rather than a length. This is useful for the loosely typed deserializer as it lets us use them for indexing. In this case though it is using the same MapAccess for strongly typed deserializing so I just pass empty names